test/services/SubmissionServiceSpec.scala

```diff
@@ -41,156 +42,298 @@ class SubmissionServiceSpec
     with MockSoleTraderMatchingService
     with MockBusinessVerificationService
     with MockCreateTrnService
+    with MockRegistrationOrchestrationService
     with FeatureSwitching {
 
-  object TestService extends SubmissionService(
-    mockJourneyService,
-    mockSoleTraderMatchingService,
-    mockSoleTraderIdentificationService,
-    mockBusinessVerificationService,
-    mockCreateTrnService)
-
   implicit val hc: HeaderCarrier = HeaderCarrier()
 
+  "submit: sole trader journey with businessVerificationCheck = false" when {
+
+    "the user has a sautr" should {
+
+      s"register Without Business Verification" in {
+
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetails, journeyConfigWithoutBV)(Future.successful(Right(true)))
+        mockRegisterWithoutBusinessVerification(testJourneyId, testIndividualDetails.optNino, testIndividualDetails.optSautr.get)(Future.successful(Registered("someId")))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl + s"?journeyId=$testJourneyId")
+
+      }
+    }
+
+    "the user has no sautr" should {
+      s"not be registered (matching does not matter)" in {
+
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        List(true, false).foreach(matchingTrueOrFalse => {
+
+          mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoSautr)))
+          mockMatchSoleTraderDetails(testJourneyId, testIndividualDetailsNoSautr, journeyConfigWithoutBV)(Future.successful(Right(matchingTrueOrFalse)))
+          mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+          val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+          result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+          verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+          resetAllMocks()
+        })
+
+      }
+    }
+
+    "the user has no Nino (possible only when EnableOptionalNinoJourney is true)" should {
+
+      s"not register but create a trn (matching does not matter)" in {
+
+        enable(EnableOptionalNinoJourney)
```

Comment:
I would disable the feature switch at the end of this test. The order in which unit tests are run is not always reliable.